### DISCUSSION TITLE
PDF not rendering

### DISCUSSION THREAD

**Discussion by salger1** [thegetty/quire#963] [Get Help]
I'm having trouble building a PDF of my catalogue in Quire 1.0.0-rc.10. The pdf.html file builds just fine and shows all of the content when I open it but when I go to build the pdf using "quire pdf" the end file only renders the first 179 pages out of the approximately 1300 total pages (see attached terminal output). 

[pdf .txt](https://github.com/user-attachments/files/17381197/pdf.txt)

**Comment by Erin-Cecele** [comment]
Hi @salger1, are you running `quire build` on a Mac or a PC? 

Can you try `quire pdf --lib prince` as an alternative and let me know if that works instead?

Thank you!

**Reply by salger1** [reply]
The Prince pdf worked but the page numbers on the subsection contents pages are all showing 0. I'm on a mac.

**Reply by Erin-Cecele** [reply]
Hmm, @salger1, looking at your Terminal output, I suspect this may be related to the TimeoutError.
<img width="834" alt="Screenshot 2024-10-15 at 2 45 52 PM" src="https://github.com/user-attachments/assets/431d15f1-986a-46ce-abae-8037e64547de">
@geealbers have we experienced a TimeoutError when building a large PDF? Any thoughts on a workaround? If not for Paged.js then at least with Prince?

**Reply by Erin-Cecele** [reply]
@cbutcosk pinging you in case you've encountered this error as well.

**Reply by cbutcosk** [reply]
Hmm alas I haven't seen it @Erin-Cecele , though...

@salger1 You might try printing pdf.html with pagedjs-cli directly with the command `npx pagedjs-cli --output publication.pdf --timeout 20000 _site/pdf.html` and just keep increasing the timeout number by 5000 until it works or something else breaks. If you find a copacetic timeout number you can alter the options passed to the printer here: https://github.com/thegetty/quire/blob/2edbb06c5e9fc9f06cfe0e31dd578adf43a73bcb/packages/cli/src/lib/pdf/paged.js#L30 (Edited to add: Erm, sorry, forgot that the CLI code is not project-specific so it's a bit hard to patch those options in a durable way)

If you'd prefer to use prince, take a look at https://github.com/thegetty/quire-starter-default/blob/e0133ece1560d1f67b7f6491ad22e81ed9e462b1/content/_assets/styles/print.scss#L212 , which attaches the page string to breaks of content within `.quire-page` elements.

**Reply by nrghamilton** [reply]
Came here today looking for a solution for same and this fixed it, thanks :)
